src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,149 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (size == 0) {
+            addToZeroPosition(value);
+        } else {
+            tail.next = new Node<T>(tail, value, null);
+            tail = tail.next;
+            size++;
+        }
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Incorrect index" + index);
+        }
+        if (index == size) {
+            add(value);
+            return;
+        } else if (index == 0) {
```

Comment:
```suggestion
        }
         if (index == 0) {
```
we don't need else keyword after return